Misc doc changes

kianmeng opened the pull request:
Besides other documentation changes, this commit ensures the generated
HTML doc for HexDocs.pm will become the source of truth for this Elixir
library and leverage on latest features of ExDoc.

kianmeng commented:
@sega-yarkin @akeemboatswain Should I resolve the conflict and resubmit the PR?

megalithic commented on lib/exlasticsearch/query.ex:
````diff
@@ -1,8 +1,10 @@
+  ```elixir
   Queryable.search_query()
   |> must(math(field, value))
   |> should(match_phrash(field, value, opts))
````
`match_phrase`

megalithic commented on lib/exlasticsearch/query.ex:
````diff
@@ -1,8 +1,10 @@
+
+  ```elixir
   Queryable.search_query()
   |> must(math(field, value))
````
`match`

milmazz commented:
@kianmeng Hello, I recently join this project as a maintainer. So, first, sorry for the long delay for an answer. Second, please let me know if you still would be interested in updating the docs? I would be happy to review and merge this PR whenever is ready.

kianmeng committed:
Misc doc changes

List of changes:
- add badges for quick links to essential module info
- add license to generated HTML doc
- bump ex_doc to latest version
- fix typo via `codespell`
- fix wordings and Markdown

kianmeng commented:
@milmazz Done. Let me know if there are anything else. Thanks.

milmazz commented:
@kianmeng Thanks! :bowtie:

kianmeng commented:
:partying_face: :partying_face: :partying_face: :partying_face: :partying_face:
